Mathlib/Data/Finite/Defs.lean

```diff
@@ -0,0 +1,135 @@
+in a computable way: it gives a concrete algorithm to produce a `Finset` whose elements enumerate
+the terms of the given type. As such, one generally relies on congruence lemmas when rewriting
+expressions involving `Fintype` instances.
+
+Every `Fintype` instance automatically gives a `Finite` instance, see `Fintype.finite`, but not vice
+versa. Every `Fintype` instance should be computable since they are meant for computation. If it's
+not possible to write a computable `Fintype` instance, one should prefer writing a `Finite` instance
+instead.
+
+## Main definitions
+
+* `Finite α` denotes that `α` is a finite type.
+* `Infinite α` denotes that `α` is an infinite type.
+
+## Implementation notes
+
+The definition of `Finite α` is not just `NonEmpty (Fintype α)` since `Fintype` requires
+that `α : Type*`, and the definition in this module allows for `α : Sort*`. This means
+we can write the instance `Finite.prop`.
+
+## Tags
+
+finite, fintype
+-/
+
+
+universe u v
+
+open Function
+
+variable {α β : Sort _}
+
+/-- A type is `Finite` if it is in bijective correspondence to some
+`Fin n`.
+
+While this could be defined as `NonEmpty (Fintype α)`, it is defined
+in this way to allow there to be `Finite` instances for propositions.
+-/
+class inductive Finite (α : Sort _) : Prop
+  | intro {n : ℕ} : α ≃ Fin n → Finite _
+#align finite Finite
+
+theorem finite_iff_exists_equiv_fin {α : Sort _} : Finite α ↔ ∃ n, Nonempty (α ≃ Fin n) :=
+  ⟨fun ⟨e⟩ => ⟨_, ⟨e⟩⟩, fun ⟨_, ⟨e⟩⟩ => ⟨e⟩⟩
+#align finite_iff_exists_equiv_fin finite_iff_exists_equiv_fin
+
+theorem Finite.exists_equiv_fin (α : Sort _) [h : Finite α] : ∃ n : ℕ, Nonempty (α ≃ Fin n) :=
+  finite_iff_exists_equiv_fin.mp h
+#align finite.exists_equiv_fin Finite.exists_equiv_fin
+
+theorem Finite.of_equiv (α : Sort _) [h : Finite α] (f : α ≃ β) : Finite β := by
+  cases' h with n e
+  exact Finite.intro (f.symm.trans e)
+#align finite.of_equiv Finite.of_equiv
+
+theorem Equiv.finite_iff (f : α ≃ β) : Finite α ↔ Finite β :=
+  ⟨fun _ => Finite.of_equiv _ f, fun _ => Finite.of_equiv _ f.symm⟩
+#align equiv.finite_iff Equiv.finite_iff
+
+theorem Function.Bijective.finite_iff {f : α → β} (h : Bijective f) : Finite α ↔ Finite β :=
+  (Equiv.ofBijective f h).finite_iff
+#align function.bijective.finite_iff Function.Bijective.finite_iff
+
+theorem Finite.ofBijective [Finite α] {f : α → β} (h : Bijective f) : Finite β :=
+  h.finite_iff.mp ‹_›
+#align finite.of_bijective Finite.ofBijective
+
+instance [Finite α] : Finite (PLift α) :=
+  Finite.of_equiv α Equiv.plift.symm
+
+instance {α : Type v} [Finite α] : Finite (ULift.{u} α) :=
+  Finite.of_equiv α Equiv.ulift.symm
+
+/-- A type is said to be infinite if it is not finite. Note that `Infinite α` is equivalent to
+`isEmpty (Fintype α)` or `isEmpty (Finite α)`. -/
+class Infinite (α : Sort _) : Prop where
+  /-- assertion that `α` is `¬Finite`-/
+  notFinite : ¬Finite α
```

Comment:
I understand the (WIP) naming convention says that structure fields should be `lowerCamelCase`, but `Prop`-valued fields are often exported as theorems, which should be `lower_snake_case`. Mathport defaults to `not_finite` here. Perhaps worth discussing on Zulip.

Reply:
I think this has been discussed, and you've heard that you were incorrect in understanding the structure fields should be lowerCamelCase. They should be cased according to their type (just as for theorems and defs).